scripts/props-table-gen/webpack-props-plugin.js

```diff
@@ -0,0 +1,194 @@
+        props.defaultValue = value.defaultValue ? Object.values(value.defaultValue)[0] : 'none';
+        props.description = marked(value.description);
+      });
+
+      // Add text wrapping to description code blocks.
+      let styledDescription;
+      if (props.description.includes('<code>')) {
+        styledDescription = props.description.replace('<code>', '<code style="white-space: pre-wrap">');
+        props.description = styledDescription;
+      }
+
+      return props;
+    };
+
+    const generateTable = (component, rowArray) => {
+      // Separates multi-word component names for prop table titles
+      const formatName = packageName(component).replace(/([A-Z])/g, ' $1').trim();
+
+      const generatedTable = ``
+      + `<div>`
+        + `<h2>${formatName} Props</h2>`
+        + `<table style="display: table">${tableHeaderTemplate}`
+          + `<tbody>${rowArray.join('')}</tbody>`
+        + `</table>`
+      + `</div>`;
+
+      return generatedTable;
+    };
+
+    // Generate props table markdown files in each component's docs folder
+    const generateMarkdownFile = (component, formattedProps) => {
+      // Creates separate name based on path, e.g. 'terra-action-header'.
+      const componentName = component.slice(0, component.indexOf('/'));
+
+      const fileName = packageName(component);
+      const fileDir = `${path.resolve(__dirname, '../../packages', componentName)}/docs/${fileName}-props-table.md`;
+
+      fs.writeFileSync(fileDir, formattedProps);
+    };
+
+    const generateMarkdown = (component, propTable) => {
+      const formattedMarkdownRows = [];
+      const propsArray = Object.entries(propTable);
+
+      // Push completely formatted table rows to an array, skipping private props.
+      propsArray.forEach((propSet) => {
+        if (propSet[1].description.includes('@private')) return;
+        formattedMarkdownRows.push(createTableRow(formatPropShape(propSet)));
+      });
+
+      // Generate complete props table
+      const completeTable = generateTable(component, formattedMarkdownRows);
+
+      return generateMarkdownFile(component, completeTable);
+    };
+
+    const processFile = (component) => {
+      const componentDir = path.resolve(__dirname, '../../packages/', component);
+      const currentFile = fs.readFileSync(componentDir, 'utf-8');
+
+      // Check if imported file has a propTypes object.
+      const propsCheck = JSON.stringify(currentFile);
+      if (!propsCheck.includes('propTypes')) return;
+
+      // Use react-docgen to extract props from the current component.
+      const parsedProps = reactDocs.parse(currentFile);
+
+      generateMarkdown(component, parsedProps.props);
+    };
+
+    // Uses fs and glob to traverse the directory tree looking for .jsx files.
+    const processPackageList = () => {
+      glob('**/*.jsx', { cwd: 'packages', ignore: ['**/terra-dev-site/**', '**/common/**', '**/tests/**'] }, (err, files) => {
+        if (err) console.log(err);
```

Comment:
can we use our Logger util for output instead?

Reply:
I will look into this!